roles/slurm/templates/notification_scripts/nodes.bash

```diff
@@ -1,23 +1,146 @@
 #!/bin/bash
 
+set -u
+set -o pipefail
+
+#
+# Global vars.
+#
+node_state_dir='/var/run/slurm/'
+node_state_file_old="${node_state_dir}/sinfo_report_old"
+node_state_file_new="${node_state_dir}/sinfo_report_new-${$}"
+send_notification='false'
+
 #
-# Compile JSON message payload.
-#
-read -r -d '' message << EOM
-{
-	"type": "mrkdwn",
-	"text": "*The _{{ slurm_cluster_name | capitalize }}_ cluster needs help*:  
-Please check and fix my nodes!  
-The \`sinfo\` command reports:
-\`\`\`
-$(sinfo -o "%P|%a|%D|%T|%N|%E" | tr \" \' | column -t -s '|')
-\`\`\`"
+##
+### Functions
+##
+#
+function thereShallBeOnlyOne() {
+	local _lock_file
+	local _lock_dir
+	_lock_file="${1}"
+	_lock_dir="$(dirname "${_lock_file}")"
+	mkdir -p "${_lock_dir}"  || logger --id="${$}" -s "Failed to create dir for lock file @ ${_lock_dir}."
+	exec 200>"${_lock_file}" || true # logger --id="${$}" -s "Failed to create FD 200>${_lock_file} for locking."
+	while ! flock -n 200; do
```

Comment:
Am I understanding this correctly: with this infinite loop, if another process is already running, this script will just keep on trying every 61 seconds ...

But if something is spawning this script, it would just make more and more processes waiting for flock to be freed

Reply:
That is correct. There is a hard Slurm limit though; from the `strigger` documentation:

> The program will be executed as the user who sets the trigger. If the program fails to terminate within 5 minutes, it will be killed along with any spawned processes.